spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/DirectMessageListenerContainer.java

```diff
@@ -985,6 +998,9 @@ void cancelConsumer(final String eventMessage) {
 		private void finalizeConsumer() {
 			RabbitUtils.setPhysicalCloseRequired(true);
 			RabbitUtils.closeChannel(getChannel());
+			if (!(getChannel() instanceof ChannelProxy)) { // clear the TL flag if the channel is not a proxy
+				RabbitUtils.isPhysicalCloseRequired();
```

Comment:
The same here: do not call `setPhysicalCloseRequired()` instead if not `ChannelProxy`